### PR TITLE
Feature: Initiate conversion of React components to TW

### PR DESCRIPTION
**Pull request by leosoaivan**
#### Because:
We want to convert our React components to use Tailwind. There is a lot to convert, and this starting point allows us to start re-using certain aspects of this conversion effort, i.e. button styles.

#### This commit
- Converts the project submission container
- Converts the modal overlay and container to TW
- Creates a button component using TW's `@apply` directive, prefixed with `top-` to avoid conflicts with Bootstrap naming scheme, i.e. `btn`.
- Recreates the modal close icon as text, since the `::before` pseudo-element is only possible in TW through its JIT.

#### Checklist
 - [ ] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/theodinproject/wiki/Contributing-Guide)?
 - [ ] You have verified the tests and linters all pass against your changes?
 - [ ] You have included automated tests for your changes where applicable?

### CODE REVIEW

File: app/javascript/components/project-submissions/components/modal.jsx [hunk 1]
```diff
@@ -2,19 +2,31 @@ import React from 'react';
...
+        <div className={`${modalBodyPosition} h-auto p-10 rounded-xl bg-white w-11/12 sm:w-4/5 md:w-3/5 lg:w-2/5`}>
           <button
-            className="link-button react-modal__close-btn"
+            className={modalCloseBtn}
             onClick={handleClose}
             type="button"
             aria-label="close"
-          />
+          >
+            x
```
Comment (KevinMulhern) [inline app/javascript/components/project-submissions/components/modal.jsx:28]: I've been testing a little in the review app. The modal close button doesn't look like it's rendering the font awesome version of the x unfortunately.

_I think_ we can use an icon element here for the font awesome icon instead 
`<i class="fa-solid fa-x"></i>`

**Review by KevinMulhern — COMMENTED** [review]
Hey @leosoaivan, this is looking great 💯 just a few very minor questions/suggestions.

Do we need to keep the commented out CSS for anything or would it be ok to delete it?

**Comment by leosoaivan** [comment]
> Hey @leosoaivan, this is looking great 💯 just a few very minor questions/suggestions.
> 
> Do we need to keep the commented out CSS for anything or would it be ok to delete it?

Yeah I wasn't sure, part of me was scared of getting rid of it, but will do!

File: tailwind.config.js [hunk 2]
```diff
@@ -34,6 +37,11 @@ module.exports = {
         'notification': 'rgba(74, 74, 74, 0.7)',
         'notification-hover': 'rgba(0, 0, 0, 1)',
         'new-notification': 'rgba(206, 151, 62, 1)',
+        'top-shadow': '0 0.05em 0.25em 0 rgba(0, 0, 0, 0.5)',
+        'light-accent': '#63BFC4',
```
[the remark this answers is not in the document]
Comment (leosoaivan) [reply tailwind.config.js:41]: 👍

File: tailwind.config.js [hunk 3]
```diff
@@ -34,6 +37,11 @@ module.exports = {
         'notification': 'rgba(74, 74, 74, 0.7)',
         'notification-hover': 'rgba(0, 0, 0, 1)',
         'new-notification': 'rgba(206, 151, 62, 1)',
+        'top-shadow': '0 0.05em 0.25em 0 rgba(0, 0, 0, 0.5)',
```
[the remark this answers is not in the document]
Comment (leosoaivan) [reply tailwind.config.js:40]: You're right, that was my initial attempt at re-creating our box-shadow. Deleted both.

File: app/javascript/components/project-submissions/components/modal.jsx (hunk 1 above)
Comment (leosoaivan) [reply app/javascript/components/project-submissions/components/modal.jsx]: Figured it out, finally!

**Pull request closed without merging**